rosplan_demos_interfaces/rosplan_interface_movebase/src/RPMoveBase.cpp

```diff
@@ -1,102 +1,126 @@
-		std::string wpID;
-		bool found = false;
-		for(size_t i=0; i<msg->parameters.size(); i++) {
-			if(0==msg->parameters[i].key.compare("to") or 0==msg->parameters[i].key.compare("w1")) {
-				wpID = msg->parameters[i].value;
-				found = true;
-			}
-		}
-		if(!found) {
-			ROS_INFO("KCL: (%s) aborting action dispatch; PDDL action missing required parameter ?to", params.name.c_str());
-			return false;
-		}
-		
-		// get pose from message store
-		std::vector< boost::shared_ptr<geometry_msgs::PoseStamped> > results;
-		if(message_store.queryNamed<geometry_msgs::PoseStamped>(wpID, results)) {
-			if(results.size()<1) {
-				ROS_INFO("KCL: (%s) aborting action dispatch; no matching wpID %s", params.name.c_str(), wpID.c_str());
-				return false;
-			}
-			if(results.size()>1)
-				ROS_INFO("KCL: (%s) multiple waypoints share the same wpID", params.name.c_str());
-
-			ROS_INFO("KCL: (%s) waiting for move_base action server to start", params.name.c_str());
-			action_client.waitForServer();
-
-			// dispatch MoveBase action
-			move_base_msgs::MoveBaseGoal goal;
-			geometry_msgs::PoseStamped &pose = *results[0];
-			goal.target_pose = pose;
-			action_client.sendGoal(goal);
-
-			bool finished_before_timeout = action_client.waitForResult();
-			if (finished_before_timeout) {
-
-				actionlib::SimpleClientGoalState state = action_client.getState();
-				ROS_INFO("KCL: (%s) action finished: %s", params.name.c_str(), state.toString().c_str());
-
-				if(state == actionlib::SimpleClientGoalState::SUCCEEDED) {
-
-					// publish feedback (achieved)
-					return true;
-
-				} else {
-
-					// clear costmaps
-					std_srvs::Empty emptySrv;
-					clear_costmaps_client.call(emptySrv);
-
-					// publish feedback (failed)
-					return false;
-				}
-			} else {
-				// timed out (failed)
-				action_client.cancelAllGoals();
-				ROS_INFO("KCL: (%s) action timed out", params.name.c_str());
-				return false;
-			}
-		} else {
-			// no KMS connection (failed)
-			ROS_INFO("KCL: (%s) aborting action dispatch; query to sceneDB failed", params.name.c_str());
-			return false;
-		}
-	}
+    // constructor
+    RPMoveBase::RPMoveBase(std::string &actionserver) : action_client_(actionserver, true) {
+
+        // create a node handle to manage communication with ROS network
+        ros::NodeHandle nh("~");
+
+        // get waypoints reference frame from param server
+        nh.param<std::string>("waypoint_frameid", waypoint_frameid_, "map");
+
+        // setup a move base clear costmap client (to be able to send clear costmap requests later on)
+        clear_costmaps_client_ = nh.serviceClient<std_srvs::Empty>("/move_base/clear_costmaps");
+    }
+
+    bool RPMoveBase::wpIDtoPoseStamped(std::string wpID, geometry_msgs::PoseStamped &result) {
+
+        ros::NodeHandle nh;
+        std::vector<double> wp;
+        if(nh.hasParam("/rosplan_demo_waypoints/" + wpID)) {
+            if(nh.getParam("/rosplan_demo_waypoints/" + wpID, wp)) {
```

Comment:
(See coment in yaml) I see you actually have a super key, here, why isn't it the same in the yaml then?

Reply:
launch file + yaml file slightly changed to enhance readability